rclcpp_action/test/test_client.cpp

```diff
@@ -480,23 +480,25 @@ TEST_F(TestClient, async_cancel_one_goal_with_callback)
   auto goal_handle = future_goal_handle.get();
   EXPECT_EQ(rclcpp_action::GoalStatus::STATUS_ACCEPTED, goal_handle->get_status());
 
-  bool cancel_response_received = false;
+  bool cancel_callback_received = false;
   auto future_cancel = action_client->async_cancel_goal(
     goal_handle,
-    [&cancel_response_received, goal_handle](
-      typename ActionGoalHandle::SharedPtr goal_handle_canceled, bool cancel_accepted) mutable
+    [&cancel_callback_received, goal_handle](
+      ActionCancelGoalResponse::SharedPtr response) mutable
     {
       if (
-        goal_handle_canceled->get_goal_id() == goal_handle->get_goal_id() &&
-        cancel_accepted)
+        ActionCancelGoalResponse::ERROR_NONE == response->return_code &&
+        goal_handle->get_goal_id() == response->goals_canceling[0].goal_id.uuid)
```

Comment:
@jacobperron I know it was already there, but shouldn't we assert the `goals_canceling` field size before accessing it?

Reply:
Done 800a2cb